akka-actor-typed/src/main/scala/akka/actor/typed/scaladsl/Actor.scala

```diff
@@ -183,6 +183,14 @@ object Actor {
 
     def onSignal(onSignal: PartialFunction[(ActorContext[T], Signal), Behavior[T]]): Behavior[T] =
       new BehaviorImpl.ImmutableBehavior(onMessage, onSignal)
+
+    def onResponse[M: ClassTag](handler: (ActorContext[T], M) ⇒ Behavior[T]): Immutable[T] = {
```

Comment:
I guess it could be confusing that `ClassTag` does not distinguish between `Option[Int]` and `Option[String]`. If you have two response clauses for the same erased class, the first clause would always be taken and it would fail with a `ClassCastException` when an element with the second element type would arrive.

Reply:
but general collection types should not be used as messages, not in typed nor in untyped

Reply:
Yes, that was just an example, it could also fail for all other type-parameterized values like the replication ones.